src/modules/dashboard/components/ActionsPage/DetailsWidget/DetailsWidgetSafeTransaction/components.tsx

```diff
@@ -0,0 +1,396 @@
+    <div className={widgetStyles.label}>
+      <FormattedMessage {...toRecipientMSG} />
+    </div>
+    <div className={widgetStyles.value}>
+      {recipient.id === 'filterValue' ? (
+        <AddressDetailsView item={recipient} isSafeItem={false} isCopyable />
+      ) : (
+        <DetailsWidgetUser
+          colony={colony}
+          walletAddress={recipient.profile.walletAddress}
+        />
+      )}
+    </div>
+  </div>
+);
+
+const renderTokenIcon = (tokenData: SafeBalanceToken) => {
+  const isERC20Token = (token: SafeBalanceToken): token is Erc20Token => {
+    if ('logoUri' in token) {
+      return true;
+    }
+
+    return false;
+  };
+
+  if (tokenData.name === 'Ether') {
+    return <Icon className={styles.ether} name="ether" title="Ether Logo" />;
+  }
+
+  if (isERC20Token(tokenData)) {
+    return (
+      <Avatar
+        avatarURL={tokenData.logoUri}
+        notSet={!tokenData.logoUri}
+        title={`${tokenData.name} token logo`}
+        placeholderIcon="circle-close"
+        seed={tokenData.address.toLowerCase()}
+      />
+    );
+  }
+
+  return (
+    <Avatar
+      notSet
+      title={tokenData.name}
+      placeholderIcon="circle-close"
+      seed={tokenData.address.toLowerCase()}
+    />
+  );
+};
+
+interface ValueProps {
+  transaction: SafeTransaction;
+  token: SafeBalanceToken;
+}
+
+export const Value = ({ transaction, token }: ValueProps) => (
+  <div className={classnames(widgetStyles.item, styles.tokenValue)}>
+    <div className={widgetStyles.label}>
+      <FormattedMessage {...valueMSG} />
+    </div>
+    {token && transaction.amount && (
+      <div className={styles.tokenContainer}>
+        {renderTokenIcon(token)}
+        <div className={widgetStyles.value}>
+          <Numeral value={transaction.amount} />
+          <span>{token.symbol}</span>
+        </div>
+      </div>
+    )}
+  </div>
+);
+
+interface TitleProps {
+  index: number | null;
+  title: MessageDescriptor;
+  isOpen: boolean;
+  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const Title = ({ index, title, isOpen, setIsOpen }: TitleProps) => (
+  <div className={classnames(widgetStyles.item, styles.title)}>
+    <div className={widgetStyles.label}>
+      {index && `${index}. `}
+      <FormattedMessage {...title} />
+    </div>
+    <div className={widgetStyles.value}>
+      <Icon
+        name={isOpen ? 'caret-up' : 'caret-down'}
+        appearance={{ size: 'extraTiny' }}
+        onClick={() => setIsOpen((val) => !val)}
+      />
+    </div>
+  </div>
+);
+
+interface InvisibleCopyableMaskedAddressProps {
+  address: string;
+  tooltipPlacement?: Placement;
+}
+
+export const InvisibleCopyableMaskedAddress = ({
+  address,
+  tooltipPlacement = 'right',
+}: InvisibleCopyableMaskedAddressProps) => (
+  <InvisibleCopyableAddress
+    address={address}
+    tooltipPlacement={tooltipPlacement}
+  >
+    <MaskedAddress address={address} />
+  </InvisibleCopyableAddress>
+);
```

Comment:
What's the criterion for a component to be in a folder like this one? As opposed to how we do it in other places? Normally we separate them, each in their own file, and only in the same file if they are really small (Some of these I wouldn't consider small and I feel like this file is too large for a shared components file, in comparison with other examples through the dapp).

It's not that I think this is technically wrong or something, I'm just partial to maintaining the file/project structure we already have and this is not the familiar way we approach this stuff.

Reply:
Can you give me some examples so I can replicate? 

I'm accustomed to seeing components only used once included in the parent file. In this case, however, that would have led to a 600+ line `DetailsWidgetSafeTransaction` file. So, I was trying to move these components into their own file while reducing the overhead of creating multiple separate files. 

For now, I've moved `InvisibleCopyableMaskedAddress` to `AddressDetailsView` since it's only being used there.

Reply:
When we think about grouping components, like these ones, good examples can be `ActionsPage`, `Dialogs`, `CreateColonyWizard`, or `Extensions`.

It's true that it can be a bit of a pain to have multiple separate files for each component but I think we have stuck to doing it this way because it makes the code neater or easier to find and read as everything is contained and in their own context. (This is at least my opinion on the matter)

Reply:
Have moved into own files